api/commands/compile.ts

```diff
@@ -14,9 +22,12 @@ export async function compile(
   try {
     // check dataformJson is valid before we try to compile
     const dataformJson = fs.readFileSync(`${compileConfig.projectDir}/dataform.json`, "utf8");
-    JSON.parse(dataformJson);
+    const checkValidity = checkDataformJsonValidity(JSON.parse(dataformJson));
```

Comment:
we can just let `checkDataformJsonValidity` throw directly; no need to return the error string and then throw that.

Reply:
Done